src/compile/legend.ts

```diff
@@ -105,40 +108,36 @@ namespace properties {
         break;
       case CIRCLE:
       case SQUARE:
-        /* tslint:disable:no-switch-case-fall-through */
-        symbols.shape = {value: mark};
-        /* tslint:enable:no-switch-case-fall-through */
-      case POINT:
+        symbols.shape = { value: mark };
         break;
+      case POINT:
       case LINE:
       case AREA:
+        // use default circle
         break;
     }
 
-    // fill or stroke
-    if (model.config().mark.filled) { // filled
-      // set stroke to transparent by default unless there is a config for stroke
-      symbols.stroke = { value: 'transparent' };
-      applyMarkConfig(symbols, model, FILL_STROKE_CONFIG);
+    applyMarkConfig(symbols, model, FILL_STROKE_CONFIG);
```

Comment:
If a fill / stroke is specified, it might override what we put in fill/stroke up there (while in marks, the config has lower precedence).